lib/pbench/test/unit/server/test_datasets_list.py

```diff
@@ -688,6 +698,61 @@ def test_filter_errors(self, monkeypatch, client, meta, error, message):
         assert e.value.http_status == error
         assert str(e.value) == message
 
+    @pytest.mark.parametrize(
+        "query,results",
+        [
+            ("global.legacy:t:bool", ["drb"]),
+            ("global.legacy:>2:int", ["fio_2"]),
+            ("global.legacy:>2023-05-01:date", []),
+        ],
```

Comment:
In the first scenario, I take it that we don't hit on `"fio_1"` or `"fio_2"` because their `global.legacy` values are ignored because they aren't boolean values (i.e., `4` is not True in this situation, and it's not False, either)...which probably makes sense...and hopefully no one will get caught by the fact that `0` and `1` are integers and _not_ boolean values.

In the third scenario, I take it that we get no hits because none of the values are coercible to dates.  In this test, that makes good sense -- neither `True` nor `4` are likely to make good dates -- but, if the value of `global.legacy` were `1682010231` or `1682010231000`, I might contemplate reconsidering.

In the second scenario, we don't hit on `"drb"` because booleans cannot be coerced to `int`'s?  I hope that doesn't bother anyone.